hail-ci-build.sh

```diff
@@ -1,6 +1,14 @@
 set -x
 source activate hail
-GRADLE_OPTS=-Xmx2048m ./gradlew testAll makeDocs archiveZip --gradle-user-home /gradle-cache
+GRADLE_OPTS=-Xmx2048m ./gradlew testAll makeDocs archiveZip --gradle-user-home /gradle-cache && \
+    pip install -U cloudtools && \
+    cluster start ci-test-$SOURCE_SHA-$TARGET_SHA \
+            --version devel \
+            --spark 2.2.0 \
+            --jar build/libs/hail-all-spark.jar \
+            --zip build/distributions/hail-python.zip && \
+    cluster submit ci-test-$SOURCE_SHA-$TARGET_SHA \
+            cluster-sanity-check.py
```

Comment:
Very nice!  I feel like your work is starting to pay off.

You need to shut the cluster down.  I'd do it with `gcloud` since that's likely to be more reliable (or both, to test cloudtools delete command?)

I'd run the whole script with `set -ex` and exit trap to clean up the cluster (see below).

Seems like you should also append a random string to the cluster name in case two instances of the same source-target are running.  Maybe use the job pod name?

Finally, if you're cancelling jobs, we might need another way to clean up (does the pod get any warning or is it summarily axed?)  Running a clean up script from the repo is one way.  Labelling clusters with job/pod info and having a service that watches k8s and cleans up abandoned clusters is another.  Another option is to have a single test cluster up all the time that multiple jobs run against (although this doesn't test cloudtools).

```
$ cat foo.sh
#!/bin/bash

set -ex

function f() {
    set +e # otherwise bar won't echo
    
    echo foo
    ls /foo/bar/baz
    echo bar
}

trap f EXIT

echo quux
ls /foo/bar/baz
echo quam
$ bash foo.sh
+ trap f EXIT
+ echo quux
quux
+ ls /foo/bar/baz
ls: cannot access '/foo/bar/baz': No such file or directory
+ f
+ set +e
+ echo foo
foo
+ ls /foo/bar/baz
ls: cannot access '/foo/bar/baz': No such file or directory
+ echo bar
bar
```